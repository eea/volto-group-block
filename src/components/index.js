export GroupBlockEdit from './manage/Blocks/Group/Edit';
export GroupBlockView from './manage/Blocks/Group/View';
export GroupBlockLayout from './manage/Blocks/Group/LayoutSchema';
