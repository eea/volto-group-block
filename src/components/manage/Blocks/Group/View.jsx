import React from 'react';
import { withBlockExtensions } from '@plone/volto/helpers';
import BodyComponent from './Body';

const View = (props) => {
  const { data } = props;
  const CustomTag = `${data.as || 'div'}`;
  const customId = data?.title
    ?.toLowerCase()
    ?.replace(/[^a-zA-Z-\s]/gi, '')
    ?.trim()
    ?.replace(/\s+/gi, '-');
  return (
    <CustomTag id={customId}>
      <BodyComponent {...props} />
    </CustomTag>
  );
};

export default withBlockExtensions(View);
