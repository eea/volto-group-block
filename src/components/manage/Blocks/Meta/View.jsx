import React from 'react';
import { RenderBlocks } from '@eeacms/volto-blocks-form/components';

export const MetaBlockView = (props) => {
  const { data } = props;
  return <RenderBlocks {...props} content={data?.data || {}} />;
};
