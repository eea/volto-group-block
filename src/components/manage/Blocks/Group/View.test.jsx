import React from 'react';
import View from './View';
import { render, screen } from '@testing-library/react';
import { RenderBlocks } from '@plone/volto/components';
import '@testing-library/jest-dom/extend-expect';

jest.mock('@plone/volto/components', () => ({
  RenderBlocks: jest.fn(() => <div>RenderBlocks</div>),
  BodyComponent: () => <div>BodyComponent</div>,
}));

jest.mock('@plone/volto/helpers', () => ({
  withBlockExtensions: jest.fn((Component) => Component),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: '/',
    hash: '',
    search: '',
  }),
}));

describe('View', () => {
  it('should render without crashing', () => {
    const props = {
      data: {},
      metadata: {},
      properties: {},
      variation: {},
    };
    render(<View {...props} />);
    expect(screen.getByText('RenderBlocks')).toBeInTheDocument();
  });

  it('renders with default tag and without crashing', () => {
    const props = {
      data: {
        variation: {},
      },
      metadata: {},
      properties: {},
      variation: {},
    };
    const { container } = render(<View {...props} />);
    expect(container.querySelector('div')).toBeInTheDocument();
  });

  it('renders with a custom tag and custom id', () => {
    const props = {
      data: {
        as: 'section',
        title: 'Test Title',
        data: { key: 'value' },
      },
      properties: {},
      variation: {},
    };
    const { container } = render(<View {...props} />);
    expect(container.querySelector('section')).toBeInTheDocument();
    expect(container.querySelector('#test-title')).toBeInTheDocument();
  });

  it('renders RenderBlocks with correct props', () => {
    const props = {
      data: {
        as: 'section',
        title: 'Test Title',
        data: { key: 'value' },
      },
      metadata: { meta: 'data' },
      properties: { prop: 'erty' },
      variation: {},
      location: {
        pathname: '/',
        search: '',
        hash: '',
      },
    };
    render(<View {...props} />);
    expect(RenderBlocks).toHaveBeenCalledWith(
      expect.objectContaining({
        metadata: props.metadata,
        content: props.data.data,
        location: props.location,
      }),
      {},
    );
  });
});
