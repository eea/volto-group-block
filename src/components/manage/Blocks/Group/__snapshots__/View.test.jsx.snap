// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View should render without crashing 1`] = `
<div>
  <div>
    RenderBlocks
  </div>
</div>
`;
